2006-01-18  Samuel Thibault  <samuel.thibault@ens-lyon.org>
	
	[BZ #2178]
	* sysdeps/mach/hurd/getsid.c (getsid): When parameter PID is zero,
	use _hurd_pid instead.

===================================================================
RCS file: /cvs/glibc/libc/sysdeps/mach/hurd/getsid.c,v
retrieving revision 1.4
retrieving revision 1.5
Index: sysdeps/mach/hurd/getsid.c
===================================================================
--- sysdeps/mach/hurd/getsid.c.orig
+++ sysdeps/mach/hurd/getsid.c
@@ -31,6 +31,9 @@
   if (pid == 0)
     pid = _hurd_pid;
 
+  if (pid == 0)
+    pid = _hurd_pid;
+
   err = __USEPORT (PROC, __proc_getsid (port, pid, &sid));
   if (err)
     return (pid_t) __hurd_fail (err);
